### ISSUE TITLE
Parser gives problem for sourcecontrolCS

### ISSUE THREAD

**Issue by joey-coleman** [symphonytool/symphony#105]
> The following bug was [originally reported](https://sourceforge.net/p/compassresearch/bugs/105/) on Sourceforge by [pglvdm](https://sourceforge.net/u/pglvdm/profile/), 2013-10-08 12:14:34.950000:

In the outline there is a comment about the parser not having populated this part and a pop-up error window arrives. In the Error Log I get:

java.net.SocketException: Connection reset
    at java.net.SocketInputStream.read(Unknown Source)
    at java.net.SocketInputStream.read(Unknown Source)
    at sun.nio.cs.StreamDecoder.readBytes(Unknown Source)
    at sun.nio.cs.StreamDecoder.implRead(Unknown Source)
    at sun.nio.cs.StreamDecoder.read(Unknown Source)
    at java.io.InputStreamReader.read(Unknown Source)
    at java.io.BufferedReader.fill(Unknown Source)
    at java.io.BufferedReader.readLine(Unknown Source)
    at java.io.BufferedReader.readLine(Unknown Source)
    at eu.compassresearch.core.interpreter.debug.messaging.MessageCommunicator.receiveMessage(MessageCommunicator.java:162)
    at eu.compassresearch.ide.interpreter.protocol.CmlCommunicationManager.receiveMessage(CmlCommunicationManager.java:128)
    at eu.compassresearch.ide.interpreter.protocol.CmlCommunicationManager.run(CmlCommunicationManager.java:228)

The model I am trying looks like: (see attached file)

and this error comes first time the syntax is correct...

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/105/#6066) by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-10-09 08:12:47.109000:
- **assigned_to**: Joey Coleman

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/105/#7cd4) by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-10-10 11:08:20.079000:
- Description has changed:

Diff:

```

--- old
+++ new
@@ -14,237 +14,6 @@
    at eu.compassresearch.ide.interpreter.protocol.CmlCommunicationManager.receiveMessage(CmlCommunicationManager.java:128)
    at eu.compassresearch.ide.interpreter.protocol.CmlCommunicationManager.run(CmlCommunicationManager.java:228)

-The model I am trying looks like:
----current model limitions
----the operations are prototypes
----no time logic, only using tock as global clock
----no real media data structure used
----stream events are not used between source and renderer
----only one source and one renderer
----no -> no time logic for
----no timeout logic for hearbeat logic
----the playig state for the renderer is not complet
----simple cout logic is used for simulating buffer logic
-
-types
--- Events from control layer:  the events set is the product of the FEP and the network layer events  
-SourceEvent :: type : (<CONNECT>|<DISCONNECT>|<PLAY>|<PAUSE>|<RESUME>|<STOPP>| <POWER_ON>|<POWER_OFF>)
-
---- need to add more complex data type representing the sound data buffers
---  the data structure must contain buffers id, creations time and present time.
-values
-
-time:nat = 0 -- our logic clock
-
-functions
-  
-channels 
-
---section for sourceControl channels/interfaces
-  sourceControlChannel:SourceEvent -- send control events to the streaming layer
-  sourceControlStatusChannel:SourceEvent -- recive status fronm the streaming layer not used right now!
-  SourceHeatbeatProcessChannel
-  RendererHeatbeatProcessChannel
-  SourceNackProcessChannel
-
---section for sourceCS channels/interfaces
-
-  sourceMainChannel --init channel for the sourceCS
-  sourceHeatbeatChannel:nat -- sending the source's heatbeat to the renderers
-
---section for RendererControl channels/interfaces
-
---section for RendererCS channels/interfaces
- rendererControlChannel:SourceEvent -- for now just use the events from the source enums, the renderer can only react on diconnnect from the control layer
- rendererDataChannel:nat           -- we dont want to model renderer events like volume control ect..., thier are not relevent for the model 
- rendererNackChannel:nat -- channel for sending nacks back to the source
- rendererheatBeatChannel:nat -- channel for sending heatbeats to the source
-
-chansets 
-SOURCE_CHANSET={|SourceNackProcessChannel,sourceHeatbeatChannel,SourceHeatbeatProcessChannel, RendererHeatbeatProcessChannel,sourceControlChannel,sourceControlStatusChannel,sourceMainChannel|}
-RENDERER_CHANSET={|rendererDataChannel,rendererNackChannel,rendererheatBeatChannel|}
-SoS_CHANSET={|rendererControlChannel,SOURCE_CHANSET,RENDERER_CHANSET|}-- could not make a chanset on this form SoS_CHANSET={|RENDERER_CHANSET,SOURCE_CHANSET|} 
-
-/*  out local clock domain just  a wrapper of tock
-  not used righ now!!!
-*/
-process SimpleClock =
-begin
-   operations
-   tick()
-   post time = time+1
-   -- local operations
-   
-   actions 
-   TickState= (tick(); Wait 10 ; TickState)  -- why the f... can't I say TickState= (tick(); Wait 10 ->TickState)
-    @ TickState
-
-end
-
--- CS model of a NMM source
-process SourceCS =
-begin
-   state
-   -- local state variables
-      n:nat --simulate data for now. will be changes to use the media buffer type and to a map of media buffers
-      timeout : nat := 10 
-      id:nat := 1 -- the id for this node
-   --   add global network delay
-   operations
-   -- local operations
-     updateData()--simulate our media data and timed ringbuffer -> change to 
-     post n = n+1
-     
-    getNackData(x:nat)i:nat -- simulate resend of buffers and the timed ringbuffer,
-    pre x<n
-    post i = x 
-     
-   actions
-   
-   --design concepts
-   -- in an main State (actions post fix with State) we run one main action (the action before the [{|x,..|}]operator) and a set of support actons all actions syncs on a channel alphabet
-   --  actions are post fix with Action
-   -- sematics = all support actions can be interupted from the main action. they skip on interuption
-   -- only the main action transit to a new main State
-   -- then a main action transit to a new man state, all sub action must be stopped == transit to Skip
-   -- support actions does not communincate with other support actions, only with the main action  
-     
-    DisConnectedState = DisConnectedAction [|{|SourceHeatbeatProcessChannel|}|]  SourceHeatBeatAction 
-    
-    DisConnectedAction = (sourceControlChannel.<CONNECT>-> SourceHeatbeatProcessChannel->ConnectedState-- no renderer is connected, but we are visable to the network, means we are sending our heartbeat
-    []sourceControlChannel.<POWER_OFF>->SourceHeatbeatProcessChannel->PowerDownState)  -- power being shout off -- should be modelled as coming from another channel
-   -- run the heartbeat in parelled (fock a child process for the SourceHeatBeat action then entering this state)
-    -- run parallel for states Disconnected ,Connected,Playing, Paused
-    SourceHeatBeatAction = (sourceHeatbeatChannel!id->Wait 4; SourceHeatBeatAction) 
-    /_\ (SourceHeatbeatProcessChannel ->Skip) -- if we go to power down state stop sending  )-- transition back to the stat the start this action-- assume a new action instans wil be started
-    
-    ConnectedState = ConnectedAction[|{|RendererHeatbeatProcessChannel,SourceHeatbeatProcessChannel|}|] RendererHeatBeatAction;SourceHeatBeatAction 
-    
-    ConnectedAction = (sourceControlChannel.<DISCONNECT>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->DisConnectedState -- control layer ask us to disconnect to the renderer
-    []sourceControlChannel.<PLAY>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->PlayingState -- start playing a new song
-    []sourceControlChannel.<POWER_OFF>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->PowerDownState)-- power being shout off -- should be modelled as coming from another channel 
-     -- observe the hearbeat from the renderer (fock new child process for the RendererHeatBeat action then entering this state)
-    
-    --run parallel for states Connected,Playing, Paused 
-    RendererHeatBeatAction = (rendererheatBeatChannel?n->RendererHeatBeatAction) --((rendererheatBeatChannel?n->RendererHeatBeat)[_ (timeout) _> Skip) --need time out logic here
-    /_\( RendererHeatbeatProcessChannel->Skip) -- (terminate the child process)
-     
-     -- state playing means bot both streaming and handling nacks from renderer
-    PlayingState =  StreamingAction [|{|SourceNackProcessChannel,RendererHeatbeatProcessChannel,SourceHeatbeatProcessChannel|}|]SourceNackHandlerAction;RendererHeatBeatAction;SourceHeatBeatAction -- fork a new child process for SourceNackHandler action then entering the playing state
-    
-    -- streaming is executed in main process (thread)
-    StreamingAction = (updateData(); rendererDataChannel!n-> Wait 2 ;StreamingAction) -- wait 2 = simulate steady production state == sending data
-    /_\ (sourceControlChannel.<DISCONNECT>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->SourceNackProcessChannel->DisConnectedState -- steady state == producing data and sending data
-    []sourceControlChannel.<STOPP>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->SourceNackProcessChannel->ConnectedState -- stop playing current song
-    []sourceControlChannel.<Pause>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->SourceNackProcessChannel->PausedState -- pause the song 
-    []sourceControlChannel.<POWER_OFF>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->SourceNackProcessChannel->PowerDownState )
-    
-    --run parallel for states Playing, Paused  --[_timeout _>)-- how to get this to run async process and get the timeout to work
-    SourceNackHandlerAction = (rendererNackChannel?x->getNackData(x);rendererDataChannel!x->SourceNackHandlerAction) --handling nack from the renderer, resend the buffer if is not to late
-    /_\ (SourceNackProcessChannel->Skip)
-   
-    PausedState = PausedAction [|{|SourceNackProcessChannel,RendererHeatbeatProcessChannel,SourceHeatbeatProcessChannel|}|]SourceNackHandlerAction;RendererHeatBeatAction;SourceHeatBeatAction -- fork a new child process for SourceNackHandler action then entering the playing state
-     
-    PausedAction =
-    sourceControlChannel.<DISCONNECT>->(DisConnectedState ||| PausedActionLeave) 
-    []sourceControlChannel.<RESUMED>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->SourceNackProcessChannel->PlayingState -- resume the current song
-    []sourceControlChannel.<STOP>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->SourceNackProcessChannel->ConnectedState -- 
-    []sourceControlChannel.<PLAY>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->SourceNackProcessChannel->PlayingState -- start playing a new song
-    []sourceControlChannel.<POWER_OFF>->RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->SourceNackProcessChannel->PowerDownState
-    
-    PausedActionLeave = (RendererHeatbeatProcessChannel->SourceHeatbeatProcessChannel->SourceNackProcessChannel->Skip) 
-    
-    PowerDownState = sourceControlChannel?x->DisConnectedState  -- no heatbeat when we are powerdown (not visable to the network) 
-    --any event on the control channel (the FEP) will wake the product up
-    
-    -- main entry point        
-   @ PowerDownState -- start up in the down mode_ simulate only the FEP is active 
-
-end
-
-process  SourceControlCS =
-begin
-   state
-   -- local state variables
-   
-   operations
-   -- local operations
-   
-   actions
-   -- actions
-   
-   @ Stop
-
-end
-
-process StreamingSourceCS = SourceCS [|SoS_CHANSET|] SourceControlCS -- need to add the SimpleClock process, shall not sync with the other processes  
-
-process RendererControlCS =
-begin
-   state
-   -- local state variables
-   
-   operations
-   -- local operations
-   
-   actions
-   -- actions
-   
-   @ Stop
-
-end
-
--- Constituent system process
---  CS model of an NMM renderer
-process RendererCS =
-begin
-   state
-   -- local state variables
-    n:nat --simulate consuming data by increasing n time based
-    dataSet: nat := 0 --  a whole buffer represting the silding window logic
-    timeout : nat := 10 
-    id:nat := 2 -- the id for this node using for heatbeat id ect..
-    heatbeatTime :nat := 0
-   
-   operations
-   addDataToBuffer(i:nat)b:bool --  adding data to our buffer, return true if we have 10 element ready to be played
-   post  b = (n = n +1 and if n - (dataSet* 10) >= 10 then dataSet = dataSet +1 and true else false )
-   
-   
-   actions
-     DisConnected = 
-     ((sourceControlChannel.CONNECT -> (heatbeatTime := time);Buffering)-- sync with the control channel of the source
-    []rendererControlChannel.POWER_OFF->PowerDown) -- power being shout off -- should be modelled as coming from another channel
-    ||| RendererHeatBeat 
-    
-    RendererHeatBeat = (rendererheatBeatChannel!id->Wait 4;RendererHeatBeat) -- send our heatbeat to the source
-    /_\(rendererControlChannel.POWER_OFF->Skip 
-    [] sourceControlChannel.DISCONNECT->Skip) -- kill this process, the state disconnected will start a new process of this type
-    
-    Buffering =  StartBuffering||| SourceHeatbeatLogic
-     
-    StartBuffering =(rendererDataChannel?x -> (if addDataToBuffer(x) then PlayingAndBuffering else StartBuffering)) -- then we have more then 10 buffers , we start playing  
-    /_\ (rendererControlChannel.POWER_OFF->PowerDown []sourceControlChannel.DISCONNECT->DisConnected ) 
-     
-    SourceHeatbeatLogic = (sourceHeatbeatChannel?id->Wait 4;SourceHeatbeatLogic) -- add time out heatbeat logic here
-    /_\( rendererControlChannel.POWER_OFF->Skip 
-    [] sourceControlChannel.DISCONNECT->Skip) 
-   
-    PlayingAndBuffering = Playing |||KeepBuffering
-    
-    KeepBuffering = (rendererDataChannel?x ->addDataToBuffer(x);KeepBuffering) 
-    /_\  rendererControlChannel.POWER_OFF->Skip [] sourceControlChannel.DISCONNECT->Skip 
-    
-    -- steady state consuming data
-    Playing =  rendererDataChannel?x -> Playing  -- add logic fro eating the data sets based on lower and upperbound logic
-    /_\ sourceControlChannel.DISCONNECT->DisConnected [] rendererControlChannel.POWER_OFF->PowerDown
-    
-    PowerDown =  rendererControlChannel?x->DisConnected -- any event on the control layer channel will wake us up
-   
-   @ PowerDown
-
-end
-
-process StreamingSoS = StreamingSourceCS [|SoS_CHANSET|]RendererCS
+The model I am trying looks like: (see attached file)

 and this error comes first time the syntax is correct...

```

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/105/#ae13) by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-10-10 11:08:21.402000:

This might not be the parser, but I'm still looking.

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/105/#2b7f) by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-10-16 13:42:52.138000:
- **status**: open --> closed

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/105/#5a22) by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-10-16 13:42:52.761000:

Outliner and pop-up behaviour fixed in [f5a17b8ad79fbeab9ce8914aecf5f62c8e4e577c]
That example has many other syntax problems, though :)
